src/js/messaging/reducers/folders.js

```diff
@@ -55,6 +57,9 @@ export default function folders(state = initialState, action) {
       return set('ui.nav.visible', !state.ui.nav.visible, state);
     case TOGGLE_MANAGED_FOLDERS:
       return set('ui.nav.foldersExpanded', !state.ui.nav.foldersExpanded, state);
+    case SET_CURRENT_FOLDER:
+      // The + forces +action.folderId to be a number
+      return set('inFolder', +action.folderId, state);
```

Comment:
I'd like to see it set as `data.currentItem.id` or something like that, as long as its associated with `currentItem`. And then when the fetch folder succeeds, the result can be `assign`ed instead of using `set`.